Lab 2.ipynb

```diff
@@ -1434,36 +446,86 @@
+    "for i in range(len(k)):\n",
+    "    if(k[i] >= Y):\n",
+    "        index = i\n",
+    "        break\n",
+    "\n",
+    "prob = np.sum(full_run[index:len(k)])\n",
+    "\n",
     "sigma = stats.norm.ppf(prob, loc = 0, scale = 1)\n",
     "\n",
-    "ax.plot(k_c / 1000, comb)"
+    "print(\"The probability of getting \" + str(Y) + \" photons is \" + str(prob) + \". The associated sigma is \" + str(sigma) + \".\")\n",
+    "print(\"This is not statistically significant.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c91f60e0-0ee4-443d-9dff-c72d9a00b50e",
+   "metadata": {},
+   "source": [
+    "# Problem 4"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "ce0ee1bd-b395-4a7c-85d2-47817ac94ad4",
+   "execution_count": 8,
+   "id": "49ee20bc-ad5b-4fcd-bd40-c018dfe20f69",
    "metadata": {},
-   "outputs": [],
-   "source": []
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The required value for 5 sigma in version 1 is 8.5.\n",
+      "The required value for 5 sigma in version 2 is 11.1.\n"
+     ]
+    }
+   ],
+   "source": [
+    "prob5sigma = 1/3.5e6\n",
+    "num_pixels = 10000\n",
+    "X = 1.7\n",
+    "# Version 1-\n",
+    "\n",
+    "det = stats.norm.ppf(1 - prob5sigma, loc = 0, scale = X)\n",
+    "\n",
+    "print(\"The required value for 5 sigma in version 1 is \" + str(round(det, 2)) + \".\")\n",
+    "\n",
+    "# Version 2\n",
+    "det_2= stats.norm.ppf(1 - prob5sigma / num_pixels, loc = 0, scale = X)\n",
+    "print(\"The required value for 5 sigma in version 2 is \" + str(round(det_2, 1)) + \".\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3e5ef6a9-ca9b-466e-878c-c963036ba820",
+   "metadata": {},
+   "source": [
+    "Using the cdf function with a given value returns the probability of getting that value. Thus, to get the version 1 required signal, we took the probability associated with 5 sigma (1 - 1/3.5e6), and used the inverse of the cdf function- the ppf function. This gave the signal value required to get a 5 sigma significance.\n",
+    "<br>\n",
+    "For version 2, the probability associated with 5 sigma is different because the distribution is scaled by the number of pixels (10,000). So in the ppf function we must now divide the probability of 5 sigma by the number of pixels.\n",
+    "<br>\n",
+    "The difference in brightness ends up being pretty small compared to the number of pixels (difference in brightness of 3 compared to 10,000 pixels). This makes sense because the distributions don't get any wider, they just get scaled vertically, so while the actual readings climb in volume massively, the rare readings are still very rare compared to the width of the distribution. If you continue to change the trials factor by more orders of magnitude, the 5-sigma sensitivity threshold will change, but likely only on the order of one order of magnitude (unless you introduce *many* more trials).\n"
```

Comment:
Nice explanation, would be good to do a bit of math on the last part to show how much the required signal increases with increasing trials factors